src/btree/bt_sync.c

```diff
@@ -388,6 +388,12 @@ __sync_page_skip(WT_SESSION_IMPL *session, WT_REF *ref, void *context, bool *ski
     if (ref->state != WT_REF_DISK)
         return (0);
 
+    /* Don't read pages into cache during startup or shutdown phase. */
+    if (F_ISSET(S2C(session), WT_CONN_RECOVERING | WT_CONN_CLOSING_TIMESTAMP)) {
+        *skipp = true;
+        return (0);
+    }
+
     /* Check whether this ref has any possible updates to be aborted. */
     if (!__wt_ref_addr_copy(session, ref, &addr))
```

Comment:
It's not directly relevant, but are there cases where we expect to fail copying address information from the address?

Reply:
It can fail only when ref->addr is NULL, otherwise, it always returns success.

Reply:
Are there cases when ref->addr is NULL, when we have a WT_REF_DSK? It would be useful to update the comment to be a bit more explicit that we don't expect this case - punt that to another ticket if you like.

Reply:
Added a comment to explain when `ref->addr` can be NULL. It is possible only for the pages with the deleted state may not have any on-disk address.